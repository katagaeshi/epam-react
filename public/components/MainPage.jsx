import React from 'react';
import RunningTitle from './RunningTitle';
import SearchPanel from './SearchPanel';
import FoundMovies from './FoundMovies';
import ErrorBoundary from './ErrorBoundary';

class MainPage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div>
        <ErrorBoundary>
          <RunningTitle />
          <SearchPanel />
          <FoundMovies />
          <RunningTitle />
        </ErrorBoundary>
      </div>
    );
  }
}

export default MainPage;
