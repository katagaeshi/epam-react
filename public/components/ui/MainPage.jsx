import React from 'react';
import PropTypes from 'prop-types';
import RunningTitle from './RunningTitle';
import SearchPanel from './../containers/SearchPanel';
import FoundMovies from './../containers/FoundMovies';
import MovieDetails from './../containers/MovieDetails';
import ErrorBoundary from './ErrorBoundary';

const MainPage = props => (
  <div>
    <ErrorBoundary>
      <RunningTitle />
      {props.activePanel === 'SearchPanel' ? <SearchPanel /> : <MovieDetails /> }
      <FoundMovies />
      <RunningTitle />
    </ErrorBoundary>
  </div>
);

MainPage.propTypes = {
  activePanel: PropTypes.string.isRequired,
};

export default MainPage;
