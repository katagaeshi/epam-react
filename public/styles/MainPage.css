.MainPage {
  display: flex;
  flex-flow: column;
}

.MainPage > * {
  flex: 1 100%;
}
