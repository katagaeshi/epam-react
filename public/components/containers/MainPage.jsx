// @flow

import { connect } from 'react-redux';
import MainPage from '../ui/MainPage';
import {
  setRedirect,
  findMovies,
} from './../../actions';

const mapStateToProps = state => ({
  activePanel: state.activePanel,
  movieDetailsId: state.movieDetails.toString(),
  redirect: state.redirect,
  searchQuery: state.searchQuery,
});

const mapDispatchToProps = dispatch => ({
  setRedirect: value => dispatch(setRedirect(value)),
  findMovies: (query) => {
    dispatch(findMovies({
      query,
      value: null,
      filter: null,
      sortOption: null,
    }));
  },
});

const Container = connect(mapStateToProps, mapDispatchToProps)(MainPage);
export default Container;
