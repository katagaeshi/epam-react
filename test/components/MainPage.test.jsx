import React from 'react';
import renderer from 'react-test-renderer';
import MainPage from './../../public/components/MainPage';

describe('MainPage', () => {
  it('Snapshot', () => {
    const tree = renderer
      .create(<MainPage />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
