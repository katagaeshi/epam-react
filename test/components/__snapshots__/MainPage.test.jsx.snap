// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage Snapshot 1`] = `
<div>
  <span
    className="RunningTitle"
  >
    netflixroulette
  </span>
  <div
    className="SearchPanel"
  >
    <span
      className="Header"
    >
      FIND YOUR MOVIE
    </span>
    <input
      className="SearchField"
      placeholder="Enter movie info here"
    />
    <div
      className="SearchFilter"
    >
      <label
        htmlFor="SearchFilter"
      >
        SEARCH BY
        <div
          className="RadioButton"
        >
          <input
            checked={true}
            id="SearchFilter-0"
            onChange={[Function]}
            type="radio"
            value="TITLE"
          />
          <label
            htmlFor="SearchFilter-0"
          >
            TITLE
          </label>
        </div>
        <div
          className="RadioButton"
        >
          <input
            checked={false}
            id="SearchFilter-1"
            onChange={[Function]}
            type="radio"
            value="GENRE"
          />
          <label
            htmlFor="SearchFilter-1"
          >
            GENRE
          </label>
        </div>
      </label>
    </div>
    <button
      className="SearchButton"
    >
      SEARCH
    </button>
  </div>
  <div>
    <span
      className="EmptyResults"
    >
      No films found
    </span>
  </div>
  <span
    className="RunningTitle"
  >
    netflixroulette
  </span>
</div>
`;
